lib/ddtrace/buffer.rb

```diff
@@ -56,5 +73,29 @@ def close
         @closed = true
       end
     end
+
+    private
+
+    def measure_accept(trace)
```

Comment:
Do you think it's worth return early here, `return unless Debug::Health.metrics.send_stats?`, to avoid performing computation (e.g. `ObjectSpace.memsize_of(trace)`) that won't be useful when stats are disable?

Reply:
I was thinking about this as well... downside to this is we have to do some kind of `enabled?` check everywhere, which seems kind of unnecessarily verbose. The metrics object should be responsible for that short circuit.

Perhaps instead we could introduce some kind of block syntax to our `Datadog::Metrics` class, e.g.:

```ruby
class Metrics
  # Executes block with self if enabled
  def measure
    return unless send_stats?
    yield(self) if block_given?
  end
end

metrics.measure do |m|
  m.increment :my_stat
end
```

Maybe it could use a better name to be more clear as to its purpose like `if_active` or `when_active`. Any thoughts?

Reply:
I think the syntax
```ruby
metrics.measure do |m|
  m.increment :my_stat
end
```
is starting to look heavy.

One alternative I suggest is how the [`Logger` interface works](https://ruby-doc.org/stdlib-2.6.3/libdoc/logger/rdoc/Logger.html#method-i-info-label-Examples):
```ruby
logger.info "Simple message"
logger.info { "Complex message: #{data}" }
```

Reply:
Thought about this too; should work for everything, unless the Dogstatsd API changes. I'll implement this instead though. Thanks for the suggestion!

Reply:
Implemented this, such that you can `metrics.count { expensive_calculation }` and lazily evaluate only if metrics are enabled.